In [1]:
#importing necessary libraries
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
#without encoding, getting utf-8 error.
laptop_data = pd.read_csv('laptops.csv', encoding='latin-1')
laptop_data.head()

Unnamed: 0 Company      Product   TypeName  Inches  \
0           1   Apple  MacBook Pro  Ultrabook    13.3   
1           2   Apple  Macbook Air  Ultrabook    13.3   
2           3      HP       250 G6   Notebook    15.6   
3           4   Apple  MacBook Pro  Ultrabook    15.4   
4           5   Apple  MacBook Pro  Ultrabook    13.3   

                     ScreenResolution                         Cpu   Ram  \
0  IPS Panel Retina Display 2560x1600        Intel Core i5 2.3GHz   8GB   
1                            1440x900        Intel Core i5 1.8GHz   8GB   
2                   Full HD 1920x1080  Intel Core i5 7200U 2.5GHz   8GB   
3  IPS Panel Retina Display 2880x1800        Intel Core i7 2.7GHz  16GB   
4  IPS Panel Retina Display 2560x1600        Intel Core i5 3.1GHz   8GB   

                Memory                           Gpu  OpSys  Weight  \
0            128GB SSD  Intel Iris Plus Graphics 640  macOS  1.37kg   
1  128GB Flash Storage        Intel HD Graphics 6000  macOS  1.34kg   
2            256GB SSD         Intel HD Graphics 620  No OS  1.86kg   
3            512GB SSD            AMD Radeon Pro 455  macOS  1.83kg   
4            256GB SSD  Intel Iris Plus Graphics 650  macOS  1.37kg   

   Price_euros  
0      1339.69  
1       898.94  
2       575.00  
3      2537.45  
4      1803.60

Name of the dataframe itself incates that the data is of laptops. Our data contains specifications of the laptops: Company name, Product name, Type of the product, Screen size in inches, Screen Resolution, CPU, System Memory (RAM), Storage Type (Memory), Graphics Processing Unit(GPU), Operating System(opSys), Weight of the product, Product Weight (in kg) and price (in euros).

In [3]:
laptop_data.shape

(1303, 13)

We don't have much more data; we have just 1303 rows and 13 columns. More data can make our prediction better.

In [4]:
print(laptop_data['Company'].unique())

['Apple' 'HP' 'Acer' 'Asus' 'Dell' 'Lenovo' 'Chuwi' 'MSI' 'Microsoft'
 'Toshiba' 'Huawei' 'Xiaomi' 'Vero' 'Razer' 'Mediacom' 'Samsung' 'Google'
 'Fujitsu' 'LG']


In [5]:
print(laptop_data['Company'].value_counts())

Dell         297
Lenovo       297
HP           274
Asus         158
Acer         103
MSI           54
Toshiba       48
Apple         21
Samsung        9
Razer          7
Mediacom       7
Microsoft      6
Xiaomi         4
Vero           4
Chuwi          3
Google         3
Fujitsu        3
LG             3
Huawei         2
Name: Company, dtype: int64


We have different 19 company's data.

In [6]:
laptop_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1303 entries, 0 to 1302
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Unnamed: 0        1303 non-null   int64  
 1   Company           1303 non-null   object 
 2   Product           1303 non-null   object 
 3   TypeName          1303 non-null   object 
 4   Inches            1303 non-null   float64
 5   ScreenResolution  1303 non-null   object 
 6   Cpu               1303 non-null   object 
 7   Ram               1303 non-null   object 
 8   Memory            1303 non-null   object 
 9   Gpu               1303 non-null   object 
 10  OpSys             1303 non-null   object 
 11  Weight            1303 non-null   object 
 12  Price_euros       1303 non-null   float64
dtypes: float64(2), int64(1), object(10)
memory usage: 132.5+ KB


We don't have any missing value. Unnamed is not useful, Ram and weight data type shows object, we need to remove 'GB' and 'kg' to make the datatype float. 

In [7]:
laptop_data.duplicated().sum()

0

we have no duplicate value. If we have duplicate value, we need to use drop_duplicates() to drop the duplicate value.

In [8]:
laptop_data.isnull().sum()

Unnamed: 0          0
Company             0
Product             0
TypeName            0
Inches              0
ScreenResolution    0
Cpu                 0
Ram                 0
Memory              0
Gpu                 0
OpSys               0
Weight              0
Price_euros         0
dtype: int64

No Missing Value

# Removing Unnecessary Columns

In [9]:
laptop_data.drop(columns=['Unnamed: 0'],inplace=True)
laptop_data.sample(10)

Company            Product            TypeName  Inches  \
104  Microsoft     Surface Laptop           Ultrabook    13.5   
771    Samsung         Notebook 9  2 in 1 Convertible    15.0   
444         HP    Omen 15-AX205na              Gaming    15.6   
969     Lenovo  IdeaPad 310-15IKB            Notebook    15.6   
900       Dell             XPS 13  2 in 1 Convertible    13.3   
392       Acer           Aspire 5            Notebook    15.6   
156     Lenovo     Yoga 520-14IKB  2 in 1 Convertible    14.0   
271       Asus          Rog Strix              Gaming    17.3   
264       Dell      Inspiron 3567            Notebook    15.6   
317     Lenovo        ThinkPad 13            Notebook    13.3   

                              ScreenResolution  \
104                      Touchscreen 2256x1504   
771            Full HD / Touchscreen 1920x1080   
444                IPS Panel Full HD 1920x1080   
969                          Full HD 1920x1080   
900           Quad HD+ / Touchscreen 3200x1800   
392                         IPS Panel 1366x768   
156  IPS Panel Full HD / Touchscreen 1920x1080   
271                          Full HD 1920x1080   
264                          Full HD 1920x1080   
317                                   1366x768   

                                      Cpu   Ram                Memory  \
104            Intel Core i5 7200U 2.5GHz   8GB             256GB SSD   
771            Intel Core i7 7500U 2.7GHz  16GB             256GB SSD   
444           Intel Core i7 7700HQ 2.8GHz   8GB  128GB SSD +  1TB HDD   
969            Intel Core i5 7200U 2.5GHz   6GB             256GB SSD   
900             Intel Core i7 7Y75 1.3GHz   8GB             512GB SSD   
392            Intel Core i5 8250U 1.6GHz  12GB               1TB HDD   
156            Intel Core i3 7100U 2.4GHz   4GB             256GB SSD   
271                   AMD Ryzen 1700 3GHz  16GB  256GB SSD +  1TB HDD   
264            Intel Core i5 7200U 2.5GHz   8GB               1TB HDD   
317  Intel Celeron Dual Core 3855U 1.6GHz   4GB    16GB Flash Storage   

                         Gpu         OpSys   Weight  Price_euros  
104    Intel HD Graphics 620  Windows 10 S  1.252kg      1340.00  
771           AMD Radeon 540    Windows 10   1.71kg      1799.00  
444  Nvidia GeForce GTX 1050    Windows 10    2.2kg      1099.00  
969     Nvidia GeForce 920MX    Windows 10    2.4kg       695.00  
900    Intel HD Graphics 615    Windows 10   1.24kg      2013.10  
392     Nvidia GeForce MX130    Windows 10    2.2kg       693.99  
156    Intel HD Graphics 620    Windows 10   1.74kg       629.00  
271        AMD Radeon RX 580    Windows 10    3.2kg      1549.00  
264       AMD Radeon R5 M430    Windows 10   2.24kg       565.00  
317    Intel HD Graphics 510     Chrome OS   1.45kg       459.90

We removed first column 'Unnamed: 0' permanently by using inplace=True.

# Cleaning 'Ram' and 'Weight' columns

In [10]:
laptop_data['Ram'] = laptop_data['Ram'].str.replace('GB', '')
laptop_data['Weight'] = laptop_data['Weight'].str.replace('kg', ' ')

In [11]:
laptop_data.head()

Company      Product   TypeName  Inches                    ScreenResolution  \
0   Apple  MacBook Pro  Ultrabook    13.3  IPS Panel Retina Display 2560x1600   
1   Apple  Macbook Air  Ultrabook    13.3                            1440x900   
2      HP       250 G6   Notebook    15.6                   Full HD 1920x1080   
3   Apple  MacBook Pro  Ultrabook    15.4  IPS Panel Retina Display 2880x1800   
4   Apple  MacBook Pro  Ultrabook    13.3  IPS Panel Retina Display 2560x1600   

                          Cpu Ram               Memory  \
0        Intel Core i5 2.3GHz   8            128GB SSD   
1        Intel Core i5 1.8GHz   8  128GB Flash Storage   
2  Intel Core i5 7200U 2.5GHz   8            256GB SSD   
3        Intel Core i7 2.7GHz  16            512GB SSD   
4        Intel Core i5 3.1GHz   8            256GB SSD   

                            Gpu  OpSys Weight  Price_euros  
0  Intel Iris Plus Graphics 640  macOS  1.37       1339.69  
1        Intel HD Graphics 6000  macOS  1.34        898.94  
2         Intel HD Graphics 620  No OS  1.86        575.00  
3            AMD Radeon Pro 455  macOS  1.83       2537.45  
4  Intel Iris Plus Graphics 650  macOS  1.37       1803.60

In [12]:
laptop_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1303 entries, 0 to 1302
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Company           1303 non-null   object 
 1   Product           1303 non-null   object 
 2   TypeName          1303 non-null   object 
 3   Inches            1303 non-null   float64
 4   ScreenResolution  1303 non-null   object 
 5   Cpu               1303 non-null   object 
 6   Ram               1303 non-null   object 
 7   Memory            1303 non-null   object 
 8   Gpu               1303 non-null   object 
 9   OpSys             1303 non-null   object 
 10  Weight            1303 non-null   object 
 11  Price_euros       1303 non-null   float64
dtypes: float64(2), object(10)
memory usage: 122.3+ KB


For better prediction, we need to change data in numeric form. We removed 'GB' and 'kg' from Ram and weight. But still the datatype of the columns is object. In order to change the datatype of this two column, we will use astype() and pass int and float respectively.

In [13]:
laptop_data['Ram'] = laptop_data['Ram'].astype('int32')
laptop_data['Weight'] = laptop_data['Weight'].astype('float32')

In [14]:
laptop_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1303 entries, 0 to 1302
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Company           1303 non-null   object 
 1   Product           1303 non-null   object 
 2   TypeName          1303 non-null   object 
 3   Inches            1303 non-null   float64
 4   ScreenResolution  1303 non-null   object 
 5   Cpu               1303 non-null   object 
 6   Ram               1303 non-null   int32  
 7   Memory            1303 non-null   object 
 8   Gpu               1303 non-null   object 
 9   OpSys             1303 non-null   object 
 10  Weight            1303 non-null   float32
 11  Price_euros       1303 non-null   float64
dtypes: float32(1), float64(2), int32(1), object(8)
memory usage: 112.1+ KB


Now, the Ram and Weight column changed to int and float respectively.

# Converting 'weight' and 'price' in desirable unit

In [15]:
laptop_data ['Weight_lb'] = laptop_data ['Weight'] * 2.205
laptop_data['$Price'] = round(laptop_data['Price_euros'] * 1.14,2)
laptop_data

Company                              Product            TypeName  Inches  \
0      Apple                          MacBook Pro           Ultrabook    13.3   
1      Apple                          Macbook Air           Ultrabook    13.3   
2         HP                               250 G6            Notebook    15.6   
3      Apple                          MacBook Pro           Ultrabook    15.4   
4      Apple                          MacBook Pro           Ultrabook    13.3   
...      ...                                  ...                 ...     ...   
1298  Lenovo                       Yoga 500-14ISK  2 in 1 Convertible    14.0   
1299  Lenovo                       Yoga 900-13ISK  2 in 1 Convertible    13.3   
1300  Lenovo                   IdeaPad 100S-14IBR            Notebook    14.0   
1301      HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon            Notebook    15.6   
1302    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)            Notebook    15.6   

                                ScreenResolution  \
0             IPS Panel Retina Display 2560x1600   
1                                       1440x900   
2                              Full HD 1920x1080   
3             IPS Panel Retina Display 2880x1800   
4             IPS Panel Retina Display 2560x1600   
...                                          ...   
1298   IPS Panel Full HD / Touchscreen 1920x1080   
1299  IPS Panel Quad HD+ / Touchscreen 3200x1800   
1300                                    1366x768   
1301                                    1366x768   
1302                                    1366x768   

                                       Cpu  Ram               Memory  \
0                     Intel Core i5 2.3GHz    8            128GB SSD   
1                     Intel Core i5 1.8GHz    8  128GB Flash Storage   
2               Intel Core i5 7200U 2.5GHz    8            256GB SSD   
3                     Intel Core i7 2.7GHz   16            512GB SSD   
4                     Intel Core i5 3.1GHz    8            256GB SSD   
...                                    ...  ...                  ...   
1298            Intel Core i7 6500U 2.5GHz    4            128GB SSD   
1299            Intel Core i7 6500U 2.5GHz   16            512GB SSD   
1300  Intel Celeron Dual Core N3050 1.6GHz    2   64GB Flash Storage   
1301            Intel Core i7 6500U 2.5GHz    6              1TB HDD   
1302  Intel Celeron Dual Core N3050 1.6GHz    4            500GB HDD   

                               Gpu       OpSys  Weight  Price_euros  \
0     Intel Iris Plus Graphics 640       macOS    1.37      1339.69   
1           Intel HD Graphics 6000       macOS    1.34       898.94   
2            Intel HD Graphics 620       No OS    1.86       575.00   
3               AMD Radeon Pro 455       macOS    1.83      2537.45   
4     Intel Iris Plus Graphics 650       macOS    1.37      1803.60   
...                            ...         ...     ...          ...   
1298         Intel HD Graphics 520  Windows 10    1.80       638.00   
1299         Intel HD Graphics 520  Windows 10    1.30      1499.00   
1300             Intel HD Graphics  Windows 10    1.50       229.00   
1301            AMD Radeon R5 M330  Windows 10    2.19       764.00   
1302             Intel HD Graphics  Windows 10    2.20       369.00   

      Weight_lb   $Price  
0       3.02085  1527.25  
1       2.95470  1024.79  
2       4.10130   655.50  
3       4.03515  2892.69  
4       3.02085  2056.10  
...         ...      ...  
1298    3.96900   727.32  
1299    2.86650  1708.86  
1300    3.30750   261.06  
1301    4.82895   870.96  
1302    4.85100   420.66  

[1303 rows x 14 columns]

We are making this model for USA. The pound is a unit of mass used in United States customary systems of measurement. We changed the weight to lb from kg, price to dollar from euro. Saved these columns permanently.

In [16]:
laptop_data.drop(['Weight', 'Price_euros'], axis=1, inplace=True)

# Fetching the information from 'ScreenResolution' column

In [17]:
laptop_data['TouchScreen'] = laptop_data['ScreenResolution'].apply(lambda x:1 if 'Touchscreen' in x else 0)
laptop_data.head()

Company      Product   TypeName  Inches                    ScreenResolution  \
0   Apple  MacBook Pro  Ultrabook    13.3  IPS Panel Retina Display 2560x1600   
1   Apple  Macbook Air  Ultrabook    13.3                            1440x900   
2      HP       250 G6   Notebook    15.6                   Full HD 1920x1080   
3   Apple  MacBook Pro  Ultrabook    15.4  IPS Panel Retina Display 2880x1800   
4   Apple  MacBook Pro  Ultrabook    13.3  IPS Panel Retina Display 2560x1600   

                          Cpu  Ram               Memory  \
0        Intel Core i5 2.3GHz    8            128GB SSD   
1        Intel Core i5 1.8GHz    8  128GB Flash Storage   
2  Intel Core i5 7200U 2.5GHz    8            256GB SSD   
3        Intel Core i7 2.7GHz   16            512GB SSD   
4        Intel Core i5 3.1GHz    8            256GB SSD   

                            Gpu  OpSys  Weight_lb   $Price  TouchScreen  
0  Intel Iris Plus Graphics 640  macOS    3.02085  1527.25            0  
1        Intel HD Graphics 6000  macOS    2.95470  1024.79            0  
2         Intel HD Graphics 620  No OS    4.10130   655.50            0  
3            AMD Radeon Pro 455  macOS    4.03515  2892.69            0  
4  Intel Iris Plus Graphics 650  macOS    3.02085  2056.10            0

In [18]:
laptop_data['IPS'] = laptop_data['ScreenResolution'].apply(lambda x:1 if 'IPS' in x else 0)
laptop_data.head()

Company      Product   TypeName  Inches                    ScreenResolution  \
0   Apple  MacBook Pro  Ultrabook    13.3  IPS Panel Retina Display 2560x1600   
1   Apple  Macbook Air  Ultrabook    13.3                            1440x900   
2      HP       250 G6   Notebook    15.6                   Full HD 1920x1080   
3   Apple  MacBook Pro  Ultrabook    15.4  IPS Panel Retina Display 2880x1800   
4   Apple  MacBook Pro  Ultrabook    13.3  IPS Panel Retina Display 2560x1600   

                          Cpu  Ram               Memory  \
0        Intel Core i5 2.3GHz    8            128GB SSD   
1        Intel Core i5 1.8GHz    8  128GB Flash Storage   
2  Intel Core i5 7200U 2.5GHz    8            256GB SSD   
3        Intel Core i7 2.7GHz   16            512GB SSD   
4        Intel Core i5 3.1GHz    8            256GB SSD   

                            Gpu  OpSys  Weight_lb   $Price  TouchScreen  IPS  
0  Intel Iris Plus Graphics 640  macOS    3.02085  1527.25            0    1  
1        Intel HD Graphics 6000  macOS    2.95470  1024.79            0    0  
2         Intel HD Graphics 620  No OS    4.10130   655.50            0    0  
3            AMD Radeon Pro 455  macOS    4.03515  2892.69            0    1  
4  Intel Iris Plus Graphics 650  macOS    3.02085  2056.10            0    1

In [19]:
laptop_data['ScreenResolution']=laptop_data['ScreenResolution'].apply(lambda x: x.split()[-1])
laptop_data.head()

Company      Product   TypeName  Inches ScreenResolution  \
0   Apple  MacBook Pro  Ultrabook    13.3        2560x1600   
1   Apple  Macbook Air  Ultrabook    13.3         1440x900   
2      HP       250 G6   Notebook    15.6        1920x1080   
3   Apple  MacBook Pro  Ultrabook    15.4        2880x1800   
4   Apple  MacBook Pro  Ultrabook    13.3        2560x1600   

                          Cpu  Ram               Memory  \
0        Intel Core i5 2.3GHz    8            128GB SSD   
1        Intel Core i5 1.8GHz    8  128GB Flash Storage   
2  Intel Core i5 7200U 2.5GHz    8            256GB SSD   
3        Intel Core i7 2.7GHz   16            512GB SSD   
4        Intel Core i5 3.1GHz    8            256GB SSD   

                            Gpu  OpSys  Weight_lb   $Price  TouchScreen  IPS  
0  Intel Iris Plus Graphics 640  macOS    3.02085  1527.25            0    1  
1        Intel HD Graphics 6000  macOS    2.95470  1024.79            0    0  
2         Intel HD Graphics 620  No OS    4.10130   655.50            0    0  
3            AMD Radeon Pro 455  macOS    4.03515  2892.69            0    1  
4  Intel Iris Plus Graphics 650  macOS    3.02085  2056.10            0    1

In [20]:
#laptop_data['ScreenResolution']=laptop_data['ScreenResolution'].str.extract('(\d\d\d\dx\d\d\d\d)')
#laptop_data.head()

ScreenResolution column contains informations like Touchscreen, IPS and resolution of the screen. By using individual lambda function, we fetch the info and saved in diffeent columns. Further, we by using split function, we split x resolution and y resolution.

In [21]:
laptop_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1303 entries, 0 to 1302
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Company           1303 non-null   object 
 1   Product           1303 non-null   object 
 2   TypeName          1303 non-null   object 
 3   Inches            1303 non-null   float64
 4   ScreenResolution  1303 non-null   object 
 5   Cpu               1303 non-null   object 
 6   Ram               1303 non-null   int32  
 7   Memory            1303 non-null   object 
 8   Gpu               1303 non-null   object 
 9   OpSys             1303 non-null   object 
 10  Weight_lb         1303 non-null   float32
 11  $Price            1303 non-null   float64
 12  TouchScreen       1303 non-null   int64  
 13  IPS               1303 non-null   int64  
dtypes: float32(1), float64(2), int32(1), int64(2), object(8)
memory usage: 132.5+ KB


In [22]:
new_df = laptop_data['ScreenResolution'].str.split('x', expand=True)
new_df

0     1
0     2560  1600
1     1440   900
2     1920  1080
3     2880  1800
4     2560  1600
...    ...   ...
1298  1920  1080
1299  3200  1800
1300  1366   768
1301  1366   768
1302  1366   768

[1303 rows x 2 columns]

In [23]:
laptop_data['X_res'] = new_df[0]
laptop_data['Y_res'] = new_df[1]

In [24]:
laptop_data.head()

Company      Product   TypeName  Inches ScreenResolution  \
0   Apple  MacBook Pro  Ultrabook    13.3        2560x1600   
1   Apple  Macbook Air  Ultrabook    13.3         1440x900   
2      HP       250 G6   Notebook    15.6        1920x1080   
3   Apple  MacBook Pro  Ultrabook    15.4        2880x1800   
4   Apple  MacBook Pro  Ultrabook    13.3        2560x1600   

                          Cpu  Ram               Memory  \
0        Intel Core i5 2.3GHz    8            128GB SSD   
1        Intel Core i5 1.8GHz    8  128GB Flash Storage   
2  Intel Core i5 7200U 2.5GHz    8            256GB SSD   
3        Intel Core i7 2.7GHz   16            512GB SSD   
4        Intel Core i5 3.1GHz    8            256GB SSD   

                            Gpu  OpSys  Weight_lb   $Price  TouchScreen  IPS  \
0  Intel Iris Plus Graphics 640  macOS    3.02085  1527.25            0    1   
1        Intel HD Graphics 6000  macOS    2.95470  1024.79            0    0   
2         Intel HD Graphics 620  No OS    4.10130   655.50            0    0   
3            AMD Radeon Pro 455  macOS    4.03515  2892.69            0    1   
4  Intel Iris Plus Graphics 650  macOS    3.02085  2056.10            0    1   

  X_res Y_res  
0  2560  1600  
1  1440   900  
2  1920  1080  
3  2880  1800  
4  2560  1600

In [25]:
laptop_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1303 entries, 0 to 1302
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Company           1303 non-null   object 
 1   Product           1303 non-null   object 
 2   TypeName          1303 non-null   object 
 3   Inches            1303 non-null   float64
 4   ScreenResolution  1303 non-null   object 
 5   Cpu               1303 non-null   object 
 6   Ram               1303 non-null   int32  
 7   Memory            1303 non-null   object 
 8   Gpu               1303 non-null   object 
 9   OpSys             1303 non-null   object 
 10  Weight_lb         1303 non-null   float32
 11  $Price            1303 non-null   float64
 12  TouchScreen       1303 non-null   int64  
 13  IPS               1303 non-null   int64  
 14  X_res             1303 non-null   object 
 15  Y_res             1303 non-null   object 
dtypes: float32(1), float64(2), int32(1), int64

In [26]:
laptop_data['X_res']=laptop_data['X_res'].astype('int')
laptop_data['Y_res']=laptop_data['Y_res'].astype('int')
laptop_data.head()

Company      Product   TypeName  Inches ScreenResolution  \
0   Apple  MacBook Pro  Ultrabook    13.3        2560x1600   
1   Apple  Macbook Air  Ultrabook    13.3         1440x900   
2      HP       250 G6   Notebook    15.6        1920x1080   
3   Apple  MacBook Pro  Ultrabook    15.4        2880x1800   
4   Apple  MacBook Pro  Ultrabook    13.3        2560x1600   

                          Cpu  Ram               Memory  \
0        Intel Core i5 2.3GHz    8            128GB SSD   
1        Intel Core i5 1.8GHz    8  128GB Flash Storage   
2  Intel Core i5 7200U 2.5GHz    8            256GB SSD   
3        Intel Core i7 2.7GHz   16            512GB SSD   
4        Intel Core i5 3.1GHz    8            256GB SSD   

                            Gpu  OpSys  Weight_lb   $Price  TouchScreen  IPS  \
0  Intel Iris Plus Graphics 640  macOS    3.02085  1527.25            0    1   
1        Intel HD Graphics 6000  macOS    2.95470  1024.79            0    0   
2         Intel HD Graphics 620  No OS    4.10130   655.50            0    0   
3            AMD Radeon Pro 455  macOS    4.03515  2892.69            0    1   
4  Intel Iris Plus Graphics 650  macOS    3.02085  2056.10            0    1   

   X_res  Y_res  
0   2560   1600  
1   1440    900  
2   1920   1080  
3   2880   1800  
4   2560   1600

In [27]:
laptop_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1303 entries, 0 to 1302
Data columns (total 16 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Company           1303 non-null   object 
 1   Product           1303 non-null   object 
 2   TypeName          1303 non-null   object 
 3   Inches            1303 non-null   float64
 4   ScreenResolution  1303 non-null   object 
 5   Cpu               1303 non-null   object 
 6   Ram               1303 non-null   int32  
 7   Memory            1303 non-null   object 
 8   Gpu               1303 non-null   object 
 9   OpSys             1303 non-null   object 
 10  Weight_lb         1303 non-null   float32
 11  $Price            1303 non-null   float64
 12  TouchScreen       1303 non-null   int64  
 13  IPS               1303 non-null   int64  
 14  X_res             1303 non-null   int32  
 15  Y_res             1303 non-null   int32  
dtypes: float32(1), float64(2), int32(3), int64

# Fetching the information from 'Cpu' column

In [28]:
def processor_type(text):
    if text.split()[0] == 'Intel':
        return 'Intel Processor'
    else:
        return 'AMD Processor'

In [29]:
laptop_data['Cpu processor'] = laptop_data['Cpu'].apply(processor_type)

'Cpu' column contains type of the processor and speed of the processor. By creating a function called process_type, we fetch processor type and creating lambda function we fetch speed of the processor.

In [30]:
laptop_data['Processor_speed_GHz'] = laptop_data['Cpu'].apply(lambda x: x.split()[-1])
laptop_data

Company                              Product            TypeName  Inches  \
0      Apple                          MacBook Pro           Ultrabook    13.3   
1      Apple                          Macbook Air           Ultrabook    13.3   
2         HP                               250 G6            Notebook    15.6   
3      Apple                          MacBook Pro           Ultrabook    15.4   
4      Apple                          MacBook Pro           Ultrabook    13.3   
...      ...                                  ...                 ...     ...   
1298  Lenovo                       Yoga 500-14ISK  2 in 1 Convertible    14.0   
1299  Lenovo                       Yoga 900-13ISK  2 in 1 Convertible    13.3   
1300  Lenovo                   IdeaPad 100S-14IBR            Notebook    14.0   
1301      HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon            Notebook    15.6   
1302    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)            Notebook    15.6   

     ScreenResolution                                   Cpu  Ram  \
0           2560x1600                  Intel Core i5 2.3GHz    8   
1            1440x900                  Intel Core i5 1.8GHz    8   
2           1920x1080            Intel Core i5 7200U 2.5GHz    8   
3           2880x1800                  Intel Core i7 2.7GHz   16   
4           2560x1600                  Intel Core i5 3.1GHz    8   
...               ...                                   ...  ...   
1298        1920x1080            Intel Core i7 6500U 2.5GHz    4   
1299        3200x1800            Intel Core i7 6500U 2.5GHz   16   
1300         1366x768  Intel Celeron Dual Core N3050 1.6GHz    2   
1301         1366x768            Intel Core i7 6500U 2.5GHz    6   
1302         1366x768  Intel Celeron Dual Core N3050 1.6GHz    4   

                   Memory                           Gpu       OpSys  \
0               128GB SSD  Intel Iris Plus Graphics 640       macOS   
1     128GB Flash Storage        Intel HD Graphics 6000       macOS   
2               256GB SSD         Intel HD Graphics 620       No OS   
3               512GB SSD            AMD Radeon Pro 455       macOS   
4               256GB SSD  Intel Iris Plus Graphics 650       macOS   
...                   ...                           ...         ...   
1298            128GB SSD         Intel HD Graphics 520  Windows 10   
1299            512GB SSD         Intel HD Graphics 520  Windows 10   
1300   64GB Flash Storage             Intel HD Graphics  Windows 10   
1301              1TB HDD            AMD Radeon R5 M330  Windows 10   
1302            500GB HDD             Intel HD Graphics  Windows 10   

      Weight_lb   $Price  TouchScreen  IPS  X_res  Y_res    Cpu processor  \
0       3.02085  1527.25            0    1   2560   1600  Intel Processor   
1       2.95470  1024.79            0    0   1440    900  Intel Processor   
2       4.10130   655.50            0    0   1920   1080  Intel Processor   
3       4.03515  2892.69            0    1   2880   1800  Intel Processor   
4       3.02085  2056.10            0    1   2560   1600  Intel Processor   
...         ...      ...          ...  ...    ...    ...              ...   
1298    3.96900   727.32            1    1   1920   1080  Intel Processor   
1299    2.86650  1708.86            1    1   3200   1800  Intel Processor   
1300    3.30750   261.06            0    0   1366    768  Intel Processor   
1301    4.82895   870.96            0    0   1366    768  Intel Processor   
1302    4.85100   420.66            0    0   1366    768  Intel Processor   

     Processor_speed_GHz  
0                 2.3GHz  
1                 1.8GHz  
2                 2.5GHz  
3                 2.7GHz  
4                 3.1GHz  
...                  ...  
1298              2.5GHz  
1299              2.5GHz  
1300              1.6GHz  
1301              2.5GHz  
1302              1.6GHz  

[1303 rows x 18 columns]

In [31]:
laptop_data['Processor_speed_GHz']=laptop_data['Processor_speed_GHz'].str.replace('GHz', '')
laptop_data

Company                              Product            TypeName  Inches  \
0      Apple                          MacBook Pro           Ultrabook    13.3   
1      Apple                          Macbook Air           Ultrabook    13.3   
2         HP                               250 G6            Notebook    15.6   
3      Apple                          MacBook Pro           Ultrabook    15.4   
4      Apple                          MacBook Pro           Ultrabook    13.3   
...      ...                                  ...                 ...     ...   
1298  Lenovo                       Yoga 500-14ISK  2 in 1 Convertible    14.0   
1299  Lenovo                       Yoga 900-13ISK  2 in 1 Convertible    13.3   
1300  Lenovo                   IdeaPad 100S-14IBR            Notebook    14.0   
1301      HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon            Notebook    15.6   
1302    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)            Notebook    15.6   

     ScreenResolution                                   Cpu  Ram  \
0           2560x1600                  Intel Core i5 2.3GHz    8   
1            1440x900                  Intel Core i5 1.8GHz    8   
2           1920x1080            Intel Core i5 7200U 2.5GHz    8   
3           2880x1800                  Intel Core i7 2.7GHz   16   
4           2560x1600                  Intel Core i5 3.1GHz    8   
...               ...                                   ...  ...   
1298        1920x1080            Intel Core i7 6500U 2.5GHz    4   
1299        3200x1800            Intel Core i7 6500U 2.5GHz   16   
1300         1366x768  Intel Celeron Dual Core N3050 1.6GHz    2   
1301         1366x768            Intel Core i7 6500U 2.5GHz    6   
1302         1366x768  Intel Celeron Dual Core N3050 1.6GHz    4   

                   Memory                           Gpu       OpSys  \
0               128GB SSD  Intel Iris Plus Graphics 640       macOS   
1     128GB Flash Storage        Intel HD Graphics 6000       macOS   
2               256GB SSD         Intel HD Graphics 620       No OS   
3               512GB SSD            AMD Radeon Pro 455       macOS   
4               256GB SSD  Intel Iris Plus Graphics 650       macOS   
...                   ...                           ...         ...   
1298            128GB SSD         Intel HD Graphics 520  Windows 10   
1299            512GB SSD         Intel HD Graphics 520  Windows 10   
1300   64GB Flash Storage             Intel HD Graphics  Windows 10   
1301              1TB HDD            AMD Radeon R5 M330  Windows 10   
1302            500GB HDD             Intel HD Graphics  Windows 10   

      Weight_lb   $Price  TouchScreen  IPS  X_res  Y_res    Cpu processor  \
0       3.02085  1527.25            0    1   2560   1600  Intel Processor   
1       2.95470  1024.79            0    0   1440    900  Intel Processor   
2       4.10130   655.50            0    0   1920   1080  Intel Processor   
3       4.03515  2892.69            0    1   2880   1800  Intel Processor   
4       3.02085  2056.10            0    1   2560   1600  Intel Processor   
...         ...      ...          ...  ...    ...    ...              ...   
1298    3.96900   727.32            1    1   1920   1080  Intel Processor   
1299    2.86650  1708.86            1    1   3200   1800  Intel Processor   
1300    3.30750   261.06            0    0   1366    768  Intel Processor   
1301    4.82895   870.96            0    0   1366    768  Intel Processor   
1302    4.85100   420.66            0    0   1366    768  Intel Processor   

     Processor_speed_GHz  
0                    2.3  
1                    1.8  
2                    2.5  
3                    2.7  
4                    3.1  
...                  ...  
1298                 2.5  
1299                 2.5  
1300                 1.6  
1301                 2.5  
1302                 1.6  

[1303 rows x 18 columns]

In [32]:
laptop_data.drop(['Cpu'], axis=1, inplace=True)
laptop_data.head()

Company      Product   TypeName  Inches ScreenResolution  Ram  \
0   Apple  MacBook Pro  Ultrabook    13.3        2560x1600    8   
1   Apple  Macbook Air  Ultrabook    13.3         1440x900    8   
2      HP       250 G6   Notebook    15.6        1920x1080    8   
3   Apple  MacBook Pro  Ultrabook    15.4        2880x1800   16   
4   Apple  MacBook Pro  Ultrabook    13.3        2560x1600    8   

                Memory                           Gpu  OpSys  Weight_lb  \
0            128GB SSD  Intel Iris Plus Graphics 640  macOS    3.02085   
1  128GB Flash Storage        Intel HD Graphics 6000  macOS    2.95470   
2            256GB SSD         Intel HD Graphics 620  No OS    4.10130   
3            512GB SSD            AMD Radeon Pro 455  macOS    4.03515   
4            256GB SSD  Intel Iris Plus Graphics 650  macOS    3.02085   

    $Price  TouchScreen  IPS  X_res  Y_res    Cpu processor  \
0  1527.25            0    1   2560   1600  Intel Processor   
1  1024.79            0    0   1440    900  Intel Processor   
2   655.50            0    0   1920   1080  Intel Processor   
3  2892.69            0    1   2880   1800  Intel Processor   
4  2056.10            0    1   2560   1600  Intel Processor   

  Processor_speed_GHz  
0                 2.3  
1                 1.8  
2                 2.5  
3                 2.7  
4                 3.1

Momory column contains type of the memory: SSD, HDD, Flash Storage and Hybrid. Also, it contains the data storage capacity in GB and TB.

# Fetching the information from 'Memory' column

In [33]:
laptop_data['Memory'].value_counts()

256GB SSD                        412
1TB HDD                          223
500GB HDD                        132
512GB SSD                        118
128GB SSD +  1TB HDD              94
128GB SSD                         76
256GB SSD +  1TB HDD              73
32GB Flash Storage                38
2TB HDD                           16
64GB Flash Storage                15
512GB SSD +  1TB HDD              14
1TB SSD                           14
256GB SSD +  2TB HDD              10
1.0TB Hybrid                       9
256GB Flash Storage                8
16GB Flash Storage                 7
32GB SSD                           6
180GB SSD                          5
128GB Flash Storage                4
512GB SSD +  2TB HDD               3
16GB SSD                           3
512GB Flash Storage                2
1TB SSD +  1TB HDD                 2
256GB SSD +  500GB HDD             2
128GB SSD +  2TB HDD               2
256GB SSD +  256GB SSD             2
512GB SSD +  256GB SSD             1
5

In [34]:
#we removed 0 after the decimal point
#we removed 'GB' in memory column, and replaced 'TB' with '000' (1TB = 1000GB)
#some cell contains more than 1 memory info, they have '+' sign between two piece of info. we split those cell and save in new_df2. 
laptop_data['Memory'] = laptop_data['Memory'].astype(str).replace('\.0', '')
laptop_data['Memory'] = laptop_data['Memory'].str.replace('GB', '')
laptop_data['Memory'] = laptop_data['Memory'].str.replace('TB', '000')
new_df2 = laptop_data['Memory'].str.split('+', expand=True)

#we are joing 0th index column of new_df2 with our original dataframe, creating new column 'para1'
#we are joing 1st index column of new_df2 with our original dataframe, creating new column 'para2'
laptop_data['para1'] = new_df2[0]
laptop_data['para2'] = new_df2[1]

laptop_data

Company                              Product            TypeName  Inches  \
0      Apple                          MacBook Pro           Ultrabook    13.3   
1      Apple                          Macbook Air           Ultrabook    13.3   
2         HP                               250 G6            Notebook    15.6   
3      Apple                          MacBook Pro           Ultrabook    15.4   
4      Apple                          MacBook Pro           Ultrabook    13.3   
...      ...                                  ...                 ...     ...   
1298  Lenovo                       Yoga 500-14ISK  2 in 1 Convertible    14.0   
1299  Lenovo                       Yoga 900-13ISK  2 in 1 Convertible    13.3   
1300  Lenovo                   IdeaPad 100S-14IBR            Notebook    14.0   
1301      HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon            Notebook    15.6   
1302    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)            Notebook    15.6   

     ScreenResolution  Ram             Memory                           Gpu  \
0           2560x1600    8            128 SSD  Intel Iris Plus Graphics 640   
1            1440x900    8  128 Flash Storage        Intel HD Graphics 6000   
2           1920x1080    8            256 SSD         Intel HD Graphics 620   
3           2880x1800   16            512 SSD            AMD Radeon Pro 455   
4           2560x1600    8            256 SSD  Intel Iris Plus Graphics 650   
...               ...  ...                ...                           ...   
1298        1920x1080    4            128 SSD         Intel HD Graphics 520   
1299        3200x1800   16            512 SSD         Intel HD Graphics 520   
1300         1366x768    2   64 Flash Storage             Intel HD Graphics   
1301         1366x768    6           1000 HDD            AMD Radeon R5 M330   
1302         1366x768    4            500 HDD             Intel HD Graphics   

           OpSys  Weight_lb   $Price  TouchScreen  IPS  X_res  Y_res  \
0          macOS    3.02085  1527.25            0    1   2560   1600   
1          macOS    2.95470  1024.79            0    0   1440    900   
2          No OS    4.10130   655.50            0    0   1920   1080   
3          macOS    4.03515  2892.69            0    1   2880   1800   
4          macOS    3.02085  2056.10            0    1   2560   1600   
...          ...        ...      ...          ...  ...    ...    ...   
1298  Windows 10    3.96900   727.32            1    1   1920   1080   
1299  Windows 10    2.86650  1708.86            1    1   3200   1800   
1300  Windows 10    3.30750   261.06            0    0   1366    768   
1301  Windows 10    4.82895   870.96            0    0   1366    768   
1302  Windows 10    4.85100   420.66            0    0   1366    768   

        Cpu processor Processor_speed_GHz              para1 para2  
0     Intel Processor                 2.3            128 SSD  None  
1     Intel Processor                 1.8  128 Flash Storage  None  
2     Intel Processor                 2.5            256 SSD  None  
3     Intel Processor                 2.7            512 SSD  None  
4     Intel Processor                 3.1            256 SSD  None  
...               ...                 ...                ...   ...  
1298  Intel Processor                 2.5            128 SSD  None  
1299  Intel Processor                 2.5            512 SSD  None  
1300  Intel Processor                 1.6   64 Flash Storage  None  
1301  Intel Processor                 2.5           1000 HDD  None  
1302  Intel Processor                 1.6            500 HDD  None  

[1303 rows x 19 columns]

In [35]:
#We create four columns Layer1HDD, Layer1SSD, Layer1Hybrid, Layer1Flash_Storage.
#If para1 contains HDD, then 1 will store in Layer1HDD column; otherwise 0. We did same with all four column.
laptop_data['Layer1HDD'] = laptop_data['para1'].apply(lambda x: 1 if 'HDD' in x else 0)
laptop_data['Layer1SSD'] = laptop_data['para1'].apply(lambda x: 1 if 'SSD' in x else 0)
laptop_data['Layer1Hybrid'] = laptop_data['para1'].apply(lambda x: 1 if 'Hybrid' in x else 0)
laptop_data['Layer1Flash_Storage'] = laptop_data['para1'].apply(lambda x: 1 if 'Flash Storage' in x else 0)

#We create another four columns Layer2HDD, Layer2SSD, Layer2Hybrid, Layer2Flash_Storage.
#If para2 contains HDD, then 1 will store in Layer2HDD column; otherwise 0. We did same with all four column.
laptop_data['para2'].fillna("0", inplace = True)
laptop_data['Layer2HDD'] = laptop_data['para2'].apply(lambda x: 1 if 'HDD' in x else 0)
laptop_data['Layer2SSD'] = laptop_data['para2'].apply(lambda x: 1 if 'SSD' in x else 0)
laptop_data['Layer2Hybrid'] = laptop_data['para2'].apply(lambda x: 1 if 'Hybrid' in x else 0)
laptop_data['Layer2Flash_Storage'] = laptop_data['para2'].apply(lambda x: 1 if 'Flash Storage' in x else 0)

laptop_data.sample(10)

Company                              Product     TypeName  Inches  \
18     Lenovo                    IdeaPad 320-15IKB     Notebook    15.6   
950        HP                         ZBook Studio  Workstation    15.6   
170    Huawei                           MateBook X    Ultrabook    13.0   
1274     Asus  X553SA-XX031T (N3050/4GB/500GB/W10)     Notebook    15.6   
36       Acer                             Aspire 3     Notebook    15.6   
799      Dell                          Vostro 5568     Notebook    15.6   
319      Acer                         TravelMate B     Notebook    11.6   
378      Asus                      VivoBook E403NA     Notebook    14.0   
576   Toshiba                      Tecra X40-D-10H    Ultrabook    14.0   
890        HP                          Probook 640     Notebook    14.0   

     ScreenResolution  Ram             Memory                    Gpu  \
18          1920x1080    8           1000 HDD   Nvidia GeForce 940MX   
950         1920x1080    8              8 SSD   Nvidia Quadro M1000M   
170         2160x1440    8            256 SSD  Intel HD Graphics 620   
1274         1366x768    4            500 HDD      Intel HD Graphics   
36           1366x768    4           1000 HDD  Intel HD Graphics 620   
799         1920x1080    8            256 SSD    AMD Radeon R5 M420X   
319          1366x768    4  128 Flash Storage  Intel HD Graphics 405   
378          1366x768    4   32 Flash Storage  Intel HD Graphics 500   
576         1920x1080   16            512 SSD  Intel HD Graphics 620   
890         1920x1080    4            128 SSD  Intel HD Graphics 620   

           OpSys  Weight_lb  ...              para1  para2  Layer1HDD  \
18         No OS    4.85100  ...           1000 HDD      0          1   
950   Windows 10    4.41000  ...              8 SSD      0          0   
170   Windows 10    2.31525  ...            256 SSD      0          0   
1274  Windows 10    4.85100  ...            500 HDD      0          1   
36         Linux    4.63050  ...           1000 HDD      0          1   
799   Windows 10    4.41000  ...            256 SSD      0          0   
319   Windows 10    3.08700  ...  128 Flash Storage      0          0   
378   Windows 10    3.30750  ...   32 Flash Storage      0          0   
576   Windows 10    2.73420  ...            512 SSD      0          0   
890   Windows 10    4.29975  ...            128 SSD      0          0   

      Layer1SSD  Layer1Hybrid Layer1Flash_Storage Layer2HDD Layer2SSD  \
18            0             0                   0         0         0   
950           1             0                   0         0         0   
170           1             0                   0         0         0   
1274          0             0                   0         0         0   
36            0             0                   0         0         0   
799           1             0                   0         0         0   
319           0             0                   1         0         0   
378           0             0                   1         0         0   
576           1             0                   0         0         0   
890           1             0                   0         0         0   

     Layer2Hybrid  Layer2Flash_Storage  
18              0                    0  
950             0                    0  
170             0                    0  
1274            0                    0  
36              0                    0  
799             0                    0  
319             0                    0  
378             0                    0  
576             0                    0  
890             0                    0  

[10 rows x 27 columns]

In [36]:
def mem_type(text):
    if text.split()[-1] == 'SSD':
        return 'SSD'
    elif text.split() [-1] == 'HDD':
        return 'HHD'
    elif text.split() [-1] == 'Storage':
        return 'Flash Storage'
    elif text.split() [-1] == 'Hybrid':
        return 'Hybrid'
    else:
        return ''

In [37]:
laptop_data['layer1Memorytype'] = laptop_data['para1'].apply(mem_type)
laptop_data['layer2Memorytype'] = laptop_data['para2'].apply(mem_type)
laptop_data['Memory_type'] = laptop_data['layer1Memorytype'] + ' ' + laptop_data['layer2Memorytype']
laptop_data.drop(columns=['layer1Memorytype', 'layer2Memorytype'], inplace=True)
laptop_data.sample(10)

Company             Product   TypeName  Inches ScreenResolution  Ram  \
257       HP      Envy 17-U275cl   Notebook    17.3        1920x1080   16   
510     Dell              XPS 13  Ultrabook    13.3        1920x1080    8   
248       HP         ProBook 430   Notebook    13.3        1920x1080    4   
373   Lenovo   IdeaPad 110-17ACL   Notebook    17.3         1600x900    4   
1165    Dell       Inspiron 7567     Gaming    15.6        3840x2160    8   
851     Asus    Rog GL502VM-DS74     Gaming    15.6        1920x1080   16   
799     Dell         Vostro 5568   Notebook    15.6        1920x1080    8   
61      Dell         Vostro 5471  Ultrabook    14.0        1920x1080    8   
114     Dell              XPS 13  Ultrabook    13.3        1920x1080    8   
163   Lenovo  Legion Y520-15IKBN     Gaming    15.6        1920x1080   16   

                   Memory                         Gpu       OpSys  Weight_lb  \
257              1000 HDD        Nvidia GeForce MX150  Windows 10    6.39450   
510               128 SSD       Intel HD Graphics 620  Windows 10    2.84445   
248               500 HDD      Intel UHD Graphics 620  Windows 10    3.28545   
373               128 SSD               AMD Radeon R5  Windows 10    5.73300   
1165  128 SSD +  1000 HDD  Nvidia GeForce GTX 1050 Ti  Windows 10    5.77710   
851   128 SSD +  1000 HDD     Nvidia GeForce GTX 1060  Windows 10    4.63050   
799               256 SSD         AMD Radeon R5 M420X  Windows 10    4.41000   
61                256 SSD      Intel UHD Graphics 620  Windows 10    3.74850   
114               256 SSD      Intel UHD Graphics 620  Windows 10    2.66805   
163               256 SSD  Nvidia GeForce GTX 1050 Ti  Windows 10    5.51250   

      ...       para2  Layer1HDD  Layer1SSD  Layer1Hybrid  \
257   ...           0          1          0             0   
510   ...           0          0          1             0   
248   ...           0          1          0             0   
373   ...           0          0          1             0   
1165  ...    1000 HDD          0          1             0   
851   ...    1000 HDD          0          1             0   
799   ...           0          0          1             0   
61    ...           0          0          1             0   
114   ...           0          0          1             0   
163   ...           0          0          1             0   

      Layer1Flash_Storage Layer2HDD Layer2SSD Layer2Hybrid  \
257                     0         0         0            0   
510                     0         0         0            0   
248                     0         0         0            0   
373                     0         0         0            0   
1165                    0         1         0            0   
851                     0         1         0            0   
799                     0         0         0            0   
61                      0         0         0            0   
114                     0         0         0            0   
163                     0         0         0            0   

     Layer2Flash_Storage  Memory_type  
257                    0         HHD   
510                    0         SSD   
248                    0         HHD   
373                    0         SSD   
1165                   0      SSD HHD  
851                    0      SSD HHD  
799                    0         SSD   
61                     0         SSD   
114                    0         SSD   
163                    0         SSD   

[10 rows x 28 columns]

In [38]:
#We removed string from para1 and para2
laptop_data['para1'] = laptop_data['para1'].str.replace('HDD', '')
laptop_data['para1'] = laptop_data['para1'].str.replace('SSD', '')
laptop_data['para1'] = laptop_data['para1'].str.replace('Hybrid', '')
laptop_data['para1'] = laptop_data['para1'].str.replace('Flash Storage', '')

laptop_data['para2'] = laptop_data['para2'].str.replace('HDD', '')
laptop_data['para2'] = laptop_data['para2'].str.replace('SSD', '')
laptop_data['para2'] = laptop_data['para2'].str.replace('Hybrid', '')
laptop_data['para2'] = laptop_data['para2'].str.replace('Flash Storage', '')

#It gives error with this code and with int32 at the next line code.
#laptop_data['para1'] = round(laptop_data['para1'], 0)
#laptop_data['para2'] = round(laptop_data['para2'], 0)

laptop_data['para1'] = laptop_data['para1'].astype('float64')
laptop_data['para2'] = laptop_data['para2'].astype('float64')
laptop_data

Company                              Product            TypeName  Inches  \
0      Apple                          MacBook Pro           Ultrabook    13.3   
1      Apple                          Macbook Air           Ultrabook    13.3   
2         HP                               250 G6            Notebook    15.6   
3      Apple                          MacBook Pro           Ultrabook    15.4   
4      Apple                          MacBook Pro           Ultrabook    13.3   
...      ...                                  ...                 ...     ...   
1298  Lenovo                       Yoga 500-14ISK  2 in 1 Convertible    14.0   
1299  Lenovo                       Yoga 900-13ISK  2 in 1 Convertible    13.3   
1300  Lenovo                   IdeaPad 100S-14IBR            Notebook    14.0   
1301      HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon            Notebook    15.6   
1302    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)            Notebook    15.6   

     ScreenResolution  Ram             Memory                           Gpu  \
0           2560x1600    8            128 SSD  Intel Iris Plus Graphics 640   
1            1440x900    8  128 Flash Storage        Intel HD Graphics 6000   
2           1920x1080    8            256 SSD         Intel HD Graphics 620   
3           2880x1800   16            512 SSD            AMD Radeon Pro 455   
4           2560x1600    8            256 SSD  Intel Iris Plus Graphics 650   
...               ...  ...                ...                           ...   
1298        1920x1080    4            128 SSD         Intel HD Graphics 520   
1299        3200x1800   16            512 SSD         Intel HD Graphics 520   
1300         1366x768    2   64 Flash Storage             Intel HD Graphics   
1301         1366x768    6           1000 HDD            AMD Radeon R5 M330   
1302         1366x768    4            500 HDD             Intel HD Graphics   

           OpSys  Weight_lb  ...  para2  Layer1HDD  Layer1SSD  Layer1Hybrid  \
0          macOS    3.02085  ...    0.0          0          1             0   
1          macOS    2.95470  ...    0.0          0          0             0   
2          No OS    4.10130  ...    0.0          0          1             0   
3          macOS    4.03515  ...    0.0          0          1             0   
4          macOS    3.02085  ...    0.0          0          1             0   
...          ...        ...  ...    ...        ...        ...           ...   
1298  Windows 10    3.96900  ...    0.0          0          1             0   
1299  Windows 10    2.86650  ...    0.0          0          1             0   
1300  Windows 10    3.30750  ...    0.0          0          0             0   
1301  Windows 10    4.82895  ...    0.0          1          0             0   
1302  Windows 10    4.85100  ...    0.0          1          0             0   

      Layer1Flash_Storage Layer2HDD Layer2SSD  Layer2Hybrid  \
0                       0         0         0             0   
1                       1         0         0             0   
2                       0         0         0             0   
3                       0         0         0             0   
4                       0         0         0             0   
...                   ...       ...       ...           ...   
1298                    0         0         0             0   
1299                    0         0         0             0   
1300                    1         0         0             0   
1301                    0         0         0             0   
1302                    0         0         0             0   

      Layer2Flash_Storage     Memory_type  
0                       0            SSD   
1                       0  Flash Storage   
2                       0            SSD   
3                       0            SSD   
4                       0            SSD   
...                   ...             ...  
1298                    0            SSD   
1299                    0

In [42]:
laptop_data['Total_memory_size'] = laptop_data['para1'] + laptop_data['para1']
laptop_data['Total_memory_size'] = laptop_data['Total_memory_size'].astype('int32')

Company                              Product            TypeName  Inches  \
0      Apple                          MacBook Pro           Ultrabook    13.3   
1      Apple                          Macbook Air           Ultrabook    13.3   
2         HP                               250 G6            Notebook    15.6   
3      Apple                          MacBook Pro           Ultrabook    15.4   
4      Apple                          MacBook Pro           Ultrabook    13.3   
...      ...                                  ...                 ...     ...   
1298  Lenovo                       Yoga 500-14ISK  2 in 1 Convertible    14.0   
1299  Lenovo                       Yoga 900-13ISK  2 in 1 Convertible    13.3   
1300  Lenovo                   IdeaPad 100S-14IBR            Notebook    14.0   
1301      HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon            Notebook    15.6   
1302    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)            Notebook    15.6   

     ScreenResolution  Ram             Memory                           Gpu  \
0           2560x1600    8            128 SSD  Intel Iris Plus Graphics 640   
1            1440x900    8  128 Flash Storage        Intel HD Graphics 6000   
2           1920x1080    8            256 SSD         Intel HD Graphics 620   
3           2880x1800   16            512 SSD            AMD Radeon Pro 455   
4           2560x1600    8            256 SSD  Intel Iris Plus Graphics 650   
...               ...  ...                ...                           ...   
1298        1920x1080    4            128 SSD         Intel HD Graphics 520   
1299        3200x1800   16            512 SSD         Intel HD Graphics 520   
1300         1366x768    2   64 Flash Storage             Intel HD Graphics   
1301         1366x768    6           1000 HDD            AMD Radeon R5 M330   
1302         1366x768    4            500 HDD             Intel HD Graphics   

           OpSys  Weight_lb  ...  Layer1HDD  Layer1SSD  Layer1Hybrid  \
0          macOS    3.02085  ...          0          1             0   
1          macOS    2.95470  ...          0          0             0   
2          No OS    4.10130  ...          0          1             0   
3          macOS    4.03515  ...          0          1             0   
4          macOS    3.02085  ...          0          1             0   
...          ...        ...  ...        ...        ...           ...   
1298  Windows 10    3.96900  ...          0          1             0   
1299  Windows 10    2.86650  ...          0          1             0   
1300  Windows 10    3.30750  ...          0          0             0   
1301  Windows 10    4.82895  ...          1          0             0   
1302  Windows 10    4.85100  ...          1          0             0   

      Layer1Flash_Storage  Layer2HDD Layer2SSD Layer2Hybrid  \
0                       0          0         0            0   
1                       1          0         0            0   
2                       0          0         0            0   
3                       0          0         0            0   
4                       0          0         0            0   
...                   ...        ...       ...          ...   
1298                    0          0         0            0   
1299                    0          0         0            0   
1300                    1          0         0            0   
1301                    0          0         0            0   
1302                    0          0         0            0   

      Layer2Flash_Storage     Memory_type  Total_memory_size  
0                       0            SSD                 256  
1                       0  Flash Storage                 256  
2                       0            SSD                 512  
3                       0            SSD                1024  
4                       0            SSD                 512  
...                   ...             ...                ...  
1298                

In [43]:
#Now we have eight columns, with 1 and 0 values. Those shows the type of the memory.
#But we want the actual storage capacity. so we multiplied all 8 columns with para1 and para2.
laptop_data['HDD'] = laptop_data['para1']*laptop_data['Layer1HDD'] + laptop_data['para2']*laptop_data['Layer2HDD']
laptop_data['SSD'] = laptop_data['para1']*laptop_data['Layer1SSD'] + laptop_data['para2']*laptop_data['Layer2SSD']
laptop_data['Hybrid'] = laptop_data['para1']*laptop_data['Layer1Hybrid'] + laptop_data['para2']*laptop_data['Layer2Hybrid']
laptop_data['Flash_storage'] = laptop_data['para1']*laptop_data['Layer1Flash_Storage'] + laptop_data['para2']*laptop_data['Layer2Flash_Storage']

laptop_data['HDD'] = round(laptop_data['HDD'], 0)
laptop_data['SSD'] = round(laptop_data['SSD'], 0)
laptop_data['Hybrid'] = round(laptop_data['Hybrid'], 0)
laptop_data['Flash_storage'] = round(laptop_data['Flash_storage'], 0)

In [44]:
laptop_data

Company                              Product            TypeName  Inches  \
0      Apple                          MacBook Pro           Ultrabook    13.3   
1      Apple                          Macbook Air           Ultrabook    13.3   
2         HP                               250 G6            Notebook    15.6   
3      Apple                          MacBook Pro           Ultrabook    15.4   
4      Apple                          MacBook Pro           Ultrabook    13.3   
...      ...                                  ...                 ...     ...   
1298  Lenovo                       Yoga 500-14ISK  2 in 1 Convertible    14.0   
1299  Lenovo                       Yoga 900-13ISK  2 in 1 Convertible    13.3   
1300  Lenovo                   IdeaPad 100S-14IBR            Notebook    14.0   
1301      HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon            Notebook    15.6   
1302    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)            Notebook    15.6   

     ScreenResolution  Ram             Memory                           Gpu  \
0           2560x1600    8            128 SSD  Intel Iris Plus Graphics 640   
1            1440x900    8  128 Flash Storage        Intel HD Graphics 6000   
2           1920x1080    8            256 SSD         Intel HD Graphics 620   
3           2880x1800   16            512 SSD            AMD Radeon Pro 455   
4           2560x1600    8            256 SSD  Intel Iris Plus Graphics 650   
...               ...  ...                ...                           ...   
1298        1920x1080    4            128 SSD         Intel HD Graphics 520   
1299        3200x1800   16            512 SSD         Intel HD Graphics 520   
1300         1366x768    2   64 Flash Storage             Intel HD Graphics   
1301         1366x768    6           1000 HDD            AMD Radeon R5 M330   
1302         1366x768    4            500 HDD             Intel HD Graphics   

           OpSys  Weight_lb  ...  Layer2HDD  Layer2SSD  Layer2Hybrid  \
0          macOS    3.02085  ...          0          0             0   
1          macOS    2.95470  ...          0          0             0   
2          No OS    4.10130  ...          0          0             0   
3          macOS    4.03515  ...          0          0             0   
4          macOS    3.02085  ...          0          0             0   
...          ...        ...  ...        ...        ...           ...   
1298  Windows 10    3.96900  ...          0          0             0   
1299  Windows 10    2.86650  ...          0          0             0   
1300  Windows 10    3.30750  ...          0          0             0   
1301  Windows 10    4.82895  ...          0          0             0   
1302  Windows 10    4.85100  ...          0          0             0   

      Layer2Flash_Storage     Memory_type Total_memory_size     HDD    SSD  \
0                       0            SSD                256     0.0  128.0   
1                       0  Flash Storage                256     0.0    0.0   
2                       0            SSD                512     0.0  256.0   
3                       0            SSD               1024     0.0  512.0   
4                       0            SSD                512     0.0  256.0   
...                   ...             ...               ...     ...    ...   
1298                    0            SSD                256     0.0  128.0   
1299                    0            SSD               1024     0.0  512.0   
1300                    0  Flash Storage                128     0.0    0.0   
1301                    0            HHD               2000  1000.0    0.0   
1302                    0            HHD               1000   500.0    0.0   

      Hybrid  Flash_storage  
0        0.0            0.0  
1        0.0          128.0  
2        0.0            0.0  
3        0.0            0.0  
4        0.0            0.0  
...      ...            ...  
1298     0.0            0.0  
1299     0.0            0.0  
1300     0.

In [45]:
#Drop unnecessory columns
laptop_data.drop(columns=['Memory', 'para1', 'para2', 'Layer1HDD', 'Layer1SSD', 'Layer1Hybrid', 'Layer1Flash_Storage', 'Layer2HDD', 'Layer2SSD', 'Layer2Hybrid', 'Layer2Flash_Storage'], inplace=True)
laptop_data.sample(10)

Company                            Product   TypeName  Inches  \
280    Lenovo                  IdeaPad 320-17IKB   Notebook    17.3   
582        HP                        ProBook 430   Notebook    13.3   
1130       HP  15-bs078cl (i7-7500U/8GB/2TB/W10)   Notebook    15.6   
892   Samsung                         Notebook 9  Ultrabook    15.0   
13       Dell                      Inspiron 3567   Notebook    15.6   
797      Dell                      Latitude 3580   Notebook    15.6   
1234    Apple                        MacBook Air  Ultrabook    11.6   
142    Lenovo                 Legion Y520-15IKBN     Gaming    15.6   
149      Acer                     Aspire A315-51   Notebook    15.6   
134        HP  15-BS078nr (i7-7500U/8GB/1TB/W10)   Notebook    15.6   

     ScreenResolution  Ram                       Gpu       OpSys  Weight_lb  \
280          1600x900    6      Nvidia GeForce 940MX  Windows 10    6.17400   
582         1920x1080    4     Intel HD Graphics 620  Windows 10    3.28545   
1130         1366x768    8     Intel HD Graphics 620  Windows 10    4.49820   
892         1920x1080    8     Intel HD Graphics 620  Windows 10    2.57985   
13          1920x1080    4        AMD Radeon R5 M430  Windows 10    4.85100   
797          1366x768    8     Intel HD Graphics 620  Windows 10    4.29975   
1234         1366x768    4    Intel HD Graphics 6000    Mac OS X    2.38140   
142         1920x1080    8  Nvidia GeForce GTX 1050M       No OS    5.29200   
149          1366x768    4     Intel HD Graphics 520  Windows 10    4.63050   
134          1366x768    8     Intel HD Graphics 620  Windows 10    4.52025   

       $Price  ...  X_res  Y_res    Cpu processor  Processor_speed_GHz  \
280    979.26  ...   1600    900  Intel Processor                  2.7   
582    819.66  ...   1920   1080  Intel Processor                  2.4   
1130   717.06  ...   1366    768  Intel Processor                  2.7   
892   1936.86  ...   1920   1080  Intel Processor                  2.7   
13     568.75  ...   1920   1080  Intel Processor                    2   
797    923.40  ...   1366    768  Intel Processor                  2.5   
1234  1093.26  ...   1366    768  Intel Processor                  1.6   
142    990.66  ...   1920   1080  Intel Processor                  2.8   
149    509.58  ...   1366    768  Intel Processor                    2   
134    681.72  ...   1366    768  Intel Processor                  2.7   

         Memory_type Total_memory_size     HDD    SSD  Hybrid  Flash_storage  
280          SSD HHD               256  1000.0  128.0     0.0            0.0  
582             SSD                256     0.0  128.0     0.0            0.0  
1130            HHD               4000  2000.0    0.0     0.0            0.0  
892             SSD                512     0.0  256.0     0.0            0.0  
13              SSD                512     0.0  256.0     0.0            0.0  
797             SSD                256     0.0  128.0     0.0            0.0  
1234  Flash Storage                512     0.0    0.0     0.0          256.0  
142             SSD                512     0.0  256.0     0.0            0.0  
149             HHD               2000  1000.0    0.0     0.0            0.0  
134             HHD               2000  1000.0    0.0     0.0            0.0  

[10 rows x 22 columns]

In [46]:
laptop_data['HDD'] = laptop_data['HDD'].astype('int32')
laptop_data['SSD'] = laptop_data['SSD'].astype('int32')
laptop_data['Hybrid'] = laptop_data['Hybrid'].astype('int32')
laptop_data['Flash_storage'] = laptop_data['Flash_storage'].astype('int32')
laptop_data.head()

Company      Product   TypeName  Inches ScreenResolution  Ram  \
0   Apple  MacBook Pro  Ultrabook    13.3        2560x1600    8   
1   Apple  Macbook Air  Ultrabook    13.3         1440x900    8   
2      HP       250 G6   Notebook    15.6        1920x1080    8   
3   Apple  MacBook Pro  Ultrabook    15.4        2880x1800   16   
4   Apple  MacBook Pro  Ultrabook    13.3        2560x1600    8   

                            Gpu  OpSys  Weight_lb   $Price  ...  X_res  Y_res  \
0  Intel Iris Plus Graphics 640  macOS    3.02085  1527.25  ...   2560   1600   
1        Intel HD Graphics 6000  macOS    2.95470  1024.79  ...   1440    900   
2         Intel HD Graphics 620  No OS    4.10130   655.50  ...   1920   1080   
3            AMD Radeon Pro 455  macOS    4.03515  2892.69  ...   2880   1800   
4  Intel Iris Plus Graphics 650  macOS    3.02085  2056.10  ...   2560   1600   

     Cpu processor  Processor_speed_GHz     Memory_type Total_memory_size HDD  \
0  Intel Processor                  2.3            SSD                256   0   
1  Intel Processor                  1.8  Flash Storage                256   0   
2  Intel Processor                  2.5            SSD                512   0   
3  Intel Processor                  2.7            SSD               1024   0   
4  Intel Processor                  3.1            SSD                512   0   

   SSD  Hybrid  Flash_storage  
0  128       0              0  
1    0       0            128  
2  256       0              0  
3  512       0              0  
4  256       0              0  

[5 rows x 22 columns]

In [47]:
laptop_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1303 entries, 0 to 1302
Data columns (total 22 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Company              1303 non-null   object 
 1   Product              1303 non-null   object 
 2   TypeName             1303 non-null   object 
 3   Inches               1303 non-null   float64
 4   ScreenResolution     1303 non-null   object 
 5   Ram                  1303 non-null   int32  
 6   Gpu                  1303 non-null   object 
 7   OpSys                1303 non-null   object 
 8   Weight_lb            1303 non-null   float32
 9   $Price               1303 non-null   float64
 10  TouchScreen          1303 non-null   int64  
 11  IPS                  1303 non-null   int64  
 12  X_res                1303 non-null   int32  
 13  Y_res                1303 non-null   int32  
 14  Cpu processor        1303 non-null   object 
 15  Processor_speed_GHz  1303 non-null   o

# Fetching the information from 'Gpu' column

In [48]:
laptop_data['Gpu'].value_counts()

Intel HD Graphics 620      281
Intel HD Graphics 520      185
Intel UHD Graphics 620      68
Nvidia GeForce GTX 1050     66
Nvidia GeForce GTX 1060     48
                          ... 
AMD Radeon R5 520            1
AMD Radeon R7                1
Intel HD Graphics 540        1
AMD Radeon 540               1
ARM Mali T860 MP4            1
Name: Gpu, Length: 110, dtype: int64

In [49]:
def gpu(text):
    if text[0] == 'AMD':
        return 'Good'
    elif text[0] == 'MSI':
        return 'Good'
    elif text[0] == 'Nvidia':
        return 'Good'
    else:
        return 'Average'

In [50]:
laptop_data['Gpu'] = laptop_data['Gpu'].apply(gpu)
laptop_data.sample(10)

Company              Product            TypeName  Inches  \
295   Lenovo    Legion Y720-15IKB              Gaming    15.6   
780     Dell         Alienware 17              Gaming    17.3   
150     Dell        Inspiron 5577              Gaming    15.6   
638   Lenovo        Thinkpad Yoga  2 in 1 Convertible    13.3   
1272  Lenovo   IdeaPad 100S-14IBR            Notebook    14.0   
7      Apple          Macbook Air           Ultrabook    13.3   
751       HP        EliteBook 850            Notebook    15.6   
1162      HP  Pavilion 15-cb003nv              Gaming    15.6   
123       HP          Probook 470            Notebook    17.3   
550   Lenovo    IdeaPad 320-15ISK            Notebook    15.6   

     ScreenResolution  Ram      Gpu       OpSys  Weight_lb   $Price  ...  \
295         1920x1080    8  Average  Windows 10   7.056000  1480.86  ...   
780         1920x1080   32  Average  Windows 10   9.746099  4091.23  ...   
150         1920x1080   16  Average  Windows 10   5.644800  1424.16  ...   
638         1920x1080    8  Average  Windows 10   3.020850  2003.46  ...   
1272         1366x768    2  Average  Windows 10   3.307500   261.06  ...   
7            1440x900    8  Average       macOS   2.954700  1320.92  ...   
751         1920x1080    8  Average  Windows 10   4.057200  2166.00  ...   
1162        1920x1080    8  Average  Windows 10   5.777100  1024.86  ...   
123         1920x1080   16  Average  Windows 10   5.512500  1448.94  ...   
550         1920x1080    4  Average  Windows 10   4.851000   568.86  ...   

      X_res  Y_res    Cpu processor  Processor_speed_GHz     Memory_type  \
295    1920   1080  Intel Processor                  2.8            HHD    
780    1920   1080  Intel Processor                  2.8         SSD HHD   
150    1920   1080  Intel Processor                  2.8            SSD    
638    1920   1080  Intel Processor                  2.7            SSD    
1272   1366    768  Intel Processor                  1.6  Flash Storage    
7      1440    900  Intel Processor                  1.8  Flash Storage    
751    1920   1080  Intel Processor                  2.4            SSD    
1162   1920   1080  Intel Processor                  2.8            HHD    
123    1920   1080  Intel Processor                  1.8            SSD    
550    1920   1080  Intel Processor                    2            SSD    

     Total_memory_size   HDD   SSD  Hybrid  Flash_storage  
295               2000  1000     0       0              0  
780               2000  1000  1000       0              0  
150               1024     0   512       0              0  
638                512     0   256       0              0  
1272               128     0     0       0             64  
7                  512     0     0       0            256  
751                512     0   256       0              0  
1162              2000  1000     0       0              0  
123               1024     0   512       0              0  
550                256     0   128       0              0  

[10 rows x 22 columns]

When you’re looking for a Graphics Card,  The best graphics cards are MSI, AMD, and Nvidia. So I considered these cards as good and others as average.

# Fetching the information from 'OpSys' column

In [51]:
laptop_data['OpSys'].value_counts()

Windows 10      1072
No OS             66
Linux             62
Windows 7         45
Chrome OS         27
macOS             13
Mac OS X           8
Windows 10 S       8
Android            2
Name: OpSys, dtype: int64

In [52]:
def operating_system(text):
    if text.split()[0] == 'Windows':
        return 'Windows'
    elif text.split()[0] == 'Linux':
        return 'Linux'
    elif text.split()[0] == 'Mac' and 'macOs':
        return 'Mac'
    elif text.split()[0] == 'Android':
        return 'Android'
    elif text.split()[0] == 'Chrome':
        return 'Chrome'
    else:
        return 'No OS'

In [53]:
laptop_data['OpSys'] = laptop_data['OpSys'].apply(operating_system)
laptop_data.sample()

Company Product   TypeName  Inches ScreenResolution  Ram      Gpu  \
181    Dell  XPS 13  Ultrabook    13.3        1920x1080    8  Average   

       OpSys  Weight_lb   $Price  ...  X_res  Y_res    Cpu processor  \
181  Windows    2.71215  1594.86  ...   1920   1080  Intel Processor   

     Processor_speed_GHz Memory_type Total_memory_size HDD  SSD  Hybrid  \
181                  1.6        SSD                512   0  256       0   

     Flash_storage  
181              0  

[1 rows x 22 columns]

and lastly, I assigned idividual operating system

In [55]:
#product name doesn't helpful so dropping it
laptop_data['Product'].value_counts()

XPS 13                                     30
Inspiron 3567                              29
250 G6                                     21
Legion Y520-15IKBN                         19
Vostro 3568                                19
                                           ..
15-bw007nv (A10-9620P/6GB/128GB/Radeon      1
SmartBook 130                               1
A541NA-GO342 (N3350/4GB/500GB/Linux)        1
17-X047na (i3-6006U/8GB/1TB/W10)            1
V330-15IKB (i5-8250U/4GB/500GB/FHD/W10)     1
Name: Product, Length: 618, dtype: int64

In [56]:
laptop_data.drop(['Product'], axis=1, inplace=True)
laptop_data

Company            TypeName  Inches ScreenResolution  Ram      Gpu  \
0      Apple           Ultrabook    13.3        2560x1600    8  Average   
1      Apple           Ultrabook    13.3         1440x900    8  Average   
2         HP            Notebook    15.6        1920x1080    8  Average   
3      Apple           Ultrabook    15.4        2880x1800   16  Average   
4      Apple           Ultrabook    13.3        2560x1600    8  Average   
...      ...                 ...     ...              ...  ...      ...   
1298  Lenovo  2 in 1 Convertible    14.0        1920x1080    4  Average   
1299  Lenovo  2 in 1 Convertible    13.3        3200x1800   16  Average   
1300  Lenovo            Notebook    14.0         1366x768    2  Average   
1301      HP            Notebook    15.6         1366x768    6  Average   
1302    Asus            Notebook    15.6         1366x768    4  Average   

        OpSys  Weight_lb   $Price  TouchScreen  ...  X_res  Y_res  \
0       No OS    3.02085  1527.25            0  ...   2560   1600   
1       No OS    2.95470  1024.79            0  ...   1440    900   
2       No OS    4.10130   655.50            0  ...   1920   1080   
3       No OS    4.03515  2892.69            0  ...   2880   1800   
4       No OS    3.02085  2056.10            0  ...   2560   1600   
...       ...        ...      ...          ...  ...    ...    ...   
1298  Windows    3.96900   727.32            1  ...   1920   1080   
1299  Windows    2.86650  1708.86            1  ...   3200   1800   
1300  Windows    3.30750   261.06            0  ...   1366    768   
1301  Windows    4.82895   870.96            0  ...   1366    768   
1302  Windows    4.85100   420.66            0  ...   1366    768   

        Cpu processor Processor_speed_GHz     Memory_type Total_memory_size  \
0     Intel Processor                 2.3            SSD                256   
1     Intel Processor                 1.8  Flash Storage                256   
2     Intel Processor                 2.5            SSD                512   
3     Intel Processor                 2.7            SSD               1024   
4     Intel Processor                 3.1            SSD                512   
...               ...                 ...             ...               ...   
1298  Intel Processor                 2.5            SSD                256   
1299  Intel Processor                 2.5            SSD               1024   
1300  Intel Processor                 1.6  Flash Storage                128   
1301  Intel Processor                 2.5            HHD               2000   
1302  Intel Processor                 1.6            HHD               1000   

       HDD  SSD  Hybrid  Flash_storage  
0        0  128       0              0  
1        0    0       0            128  
2        0  256       0              0  
3        0  512       0              0  
4        0  256       0              0  
...    ...  ...     ...            ...  
1298     0  128       0              0  
1299     0  512       0              0  
1300     0    0       0             64  
1301  1000    0       0              0  
1302   500    0       0              0  

[1303 rows x 21 columns]

In [57]:
#added 2 new columns